osquery/remote/transports/tls.cpp

```diff
@@ -168,20 +169,19 @@ static auto getClient() {
   std::shared_ptr<http::Client> client = nullptr;
   if (FLAGS_tls_session_reuse) {
     thread_local std::shared_ptr<http::Client> tl_client;
+    thread_local auto last_time_reseted = std::chrono::system_clock::now();
     client = tl_client;
 
     if (client.get() == nullptr) {
       tl_client = client = std::make_shared<http::Client>();
+    }
 
-      if (FLAGS_tls_session_timeout > 0) {
-        thread_local boost::asio::deadline_timer tl_timer(IOContext::get());
-
-        tl_timer.expires_from_now(
-            boost::posix_time::seconds(FLAGS_tls_session_timeout));
-        auto this_client = &tl_client;
-        tl_timer.async_wait([this_client](boost_system::error_code const&) {
-          (*this_client).reset();
-        });
+    if (FLAGS_tls_session_timeout > 0) {
+      if (std::chrono::duration_cast<std::chrono::seconds>(
```

Comment:
Does this need two `if` statements? Can this line be rolled into the one above it?
```
if (FLAGS_tls_session_timeout > 0 && std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now() - last_time_reseted).count() > FLAGS_tls_session_timeout) {
  ...
}
```
I know it doesn't quite look as nice, but I think it might be better than an `if` followed immediately by another `if`

Reply:
agree. Thanks